src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -166,7 +166,16 @@ class ProcessApiServiceImpl(
   ): PartyManagementDependency.Relationship => Future[(Option[(String, Seq[Contact])], OnboardingData)] =
     relationship => {
       for {
-        institution <- partyManagementService.retrieveInstitution(relationship.to)(bearer)
+        institution   <- partyManagementService.retrieveInstitution(relationship.to)(bearer)
+        managers      <- partyManagementService.retrieveRelationships(
+          from = None,
+          to = Some(institution.id),
+          roles = Seq(PartyManagementDependency.PartyRole.MANAGER),
+          states = Seq(PartyManagementDependency.RelationshipState.ACTIVE),
+          products = Seq.empty,
+          productRoles = Seq.empty
+        )(bearer)
+        activeManager <- extractActiveManager(managers, relationship.product.id).toFuture(ManagerNotFoundError)
```

Comment:
you could filter product directly in the `retrieveRelationships` query to avoid receiving unnecessary data (and in a future, avoid dealing with possible pagination)

Reply:
that's right :)
fixed ;)